include/pybind11/detail/internals.h

```diff
@@ -211,6 +211,54 @@ struct internals {
 #endif
 };
 
+internals &get_internals();
+
+// the internals struct (above) is shared between all the modules. local_internals are only
+// for a single module. Any changes made to internals may require an update to
+// PYBIND11_INTERNALS_VERSION, breaking backwards compatibility. local_internals is, by design,
+// restricted to a single module. Whether a module has local internals or not should not
+// impact any other modules, because the only things accessing the local internals is the
+// module that contains them.
+struct local_internals {
```

Comment:
Just moved `struct local_internals` right below `struct internals` since both types are needed in `apply_exception_translators()`. As a side effect I personally found the file easier to read when both structs where nearby.